src/how-tos/mqtt/mqtt.kafka.broker.md

```diff
@@ -0,0 +1,201 @@
+
+::: details Detailed prerequisites
+
+- A connection to the internet
+- Docker version 1.13.0+ or later is installed and running
+- Docker Desktop or Docker Desktop for Windows on WSL 2
+- Container host resources: 1 CPU, 1GB memory
+
+Optional:
+
+- Kafka 3.0+ hosted with the Docker network allowed to communicate
+- Helm 3.0+
+- Kubernetes 1.13.0+
+
+:::
+
+### Check the Kafka topics
+
+Run the docker command under the `Verify the Kafka topics created` section of the script output. Verify these topics are listed. Read more on the data in these topics in [the overview](../../concepts/kafka-proxies/mqtt-proxy.html#step-2-pub-sub-message-reflect-with-kafka).
+
+```output:no-line-numbers
+mqtt-messages
+mqtt-retained
+mqtt-sessions
+```
+
+### Listen for messages
+
+Run the docker command under the `Start a topic consumer to listen for messages` section of the script output. If you didn't use your own Kafka, you can also see all the topics in the [Kafka UI](http://localhost:8080/ui/clusters/local/all-topics).
+
+### Send a greeting
+
+Using [eclipse-mosquitto](https://hub.docker.com/_/eclipse-mosquitto) subscribe to the `zilla` topic.
+
+```bash:no-line-numbers
+docker run -it --rm eclipse-mosquitto \
+mosquitto_sub -V 'mqttv5' --topic 'zilla' \
+--host 'host.docker.internal' --port 7183 --debug
+```
+
+In a separate session, publish a message on the `zilla` topic.
+
+```bash:no-line-numbers
+docker run -it --rm eclipse-mosquitto \
+mosquitto_pub -V 'mqttv5' --topic 'zilla' --message 'Hello, world' \
+--host 'host.docker.internal' --port 7183 --debug --insecure
+```
+
+Send messages with the retained flag.
+
+```bash:no-line-numbers
+docker run -it --rm eclipse-mosquitto \
+mosquitto_pub -V 'mqttv5' --topic 'zilla' --message 'Hello, retained' --retain \
+--host 'host.docker.internal' --port 7183 --debug --insecure
+```
+
+Then restart the `mosquitto_sub` above. The latest retained message is delivered, and the other messages are not.
+
+## Creating this example yourself
+
+### Start a Kafka instance
+
+You can use your own Kafka or set up a local Kafka with [resource.kafka.compose](https://github.com/aklivity/zilla-examples/releases/latest/download/resource.kafka.compose.tar.gz) and follow the setup instructions in the `README.md`.
+
+Export these env variables or overwrite them with your remote Kafka if you skipped the local setup.
+
+```output:no-line-numbers
+export KAFKA_HOST=host.docker.internal
+export KAFKA_PORT=29092
+```
+
+### Bootstrap Kafka
+
+Create these topics in the Kafka environment.
+
+```bash:no-line-numbers
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-sessions
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-messages --config cleanup.policy=compact
+/bin/kafka-topics.sh --bootstrap-server $KAFKA_HOST:$KAFKA_PORT --create --if-not-exists --topic mqtt-retained --config cleanup.policy=compact
```

Comment:
FYI - the rendering of this puts the bash prompt before the first line only

Reply:
That is intentional. If you copy the block of text and paste it in the terminal, it looks the same. If we want the reader to copy commands separately, we can put them in separate code blocks.

Reply:
If that's the intention it would be better to put line continuation characters at the end of the first 2 lines, also indentation looks slightly off due to missing prompt for other lines.

Reply:
It is a multiline command that can be copied and run all in one go from the Kafka host or pasted into a deployment config. I added a line continuation character at the beginning so all of the commands line up.